code-studio/src/js/components/progress/progress_dot.jsx

```diff
@@ -147,12 +147,13 @@ export const ProgressDot = React.createClass({
 
   render() {
     const level = this.props.level;
-    const uid = level.uid || level.id.toString();
+    const currentId = Number(this.props.currentLevelId);
```

Comment:
`currentLevelId` is actually a UID (string) since it can contain underscores for `LevelGroup` dots.  E.g. if the `LevelGroup` has ID 1234, the pages will have UID `1234_0`, `1234_1`, `1234_2`.